zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java

```diff
@@ -141,8 +143,10 @@ public static X509AuthenticationConfig getInstance() {
   //creation of multiple objects; not using here for the consideration of read performance
   private Set<String> openReadAccessPathPrefixes;
   private Set<String> crossDomainAccessDomains;
+  private Set<String> znodeGroupAclSuperUserIds;
```

Comment:
Do we want both znodeGroupAclSuperUserId as well as znodeGroupAclSuperUserIds? Should we change old setter method with new setter method which maintains set of super users?

Reply:
TFTR, the setter and getter have different targeted objects. The setter is for the raw string that is read from the zookeeper config / system properties (it is used in QuorumPeerConfig.parseProperties, and X509AuthenticationConfig.loadSuperUserIds). And the getter is for the set of super user ids.